source/material_model/simple.cc

```diff
@@ -30,32 +29,53 @@ namespace aspect
   {
 
     template <int dim>
-    double
+    void
     Simple<dim>::
-    viscosity (const double temperature,
-               const double,
-               const std::vector<double> &composition,
-               const SymmetricTensor<2,dim> &,
-               const Point<dim> &) const
+    evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
+             MaterialModel::MaterialModelOutputs<dim> &out) const
     {
-      const double delta_temp = temperature-reference_T;
-      const double temperature_dependence = (reference_T > 0
-                                             ?
-                                             std::max(std::min(std::exp(-thermal_viscosity_exponent*delta_temp/reference_T),
-                                                               1e2),
-                                                      1e-2)
-                                             :
-                                             1.0);
-
-      double composition_dependence = 1.0;
-      if ((composition_viscosity_prefactor != 1.0) && (composition.size() > 0))
+      for (unsigned int i=0; i < in.position.size(); ++i)
         {
-          //geometric interpolation
-          return (pow(10, ((1-composition[0]) * log10(eta*temperature_dependence)
-                           + composition[0] * log10(eta*composition_viscosity_prefactor*temperature_dependence))));
+          const double delta_temp = in.temperature[i]-reference_T;
+          const double temperature_dependence = (reference_T > 0
+                                                 ?
+                                                 std::max(std::min(std::exp(-thermal_viscosity_exponent*delta_temp/reference_T),
+                                                                   1e2),
+                                                          1e-2)
+                                                 :
+                                                 1.0);
+
+          double composition_dependence = 1.0;
+          out.viscosities[i] = ((composition_viscosity_prefactor != 1.0) && (in.composition[i].size() > 0))
```

Comment:
That's probably a good idea, but it will change functionality and doesn't belong in this commit.